src/main/webapp/js/finscholar/scholarshiplistview/scholarshiplistview.js

```diff
@@ -15,21 +15,30 @@
 /** @fileoverview The mini controller for scholarship list view. */
 
 goog.module('finscholar.scholarshiplistview');
-const {scholarshiplist} = goog.require('finscholar.scholarshiplistview.templates');
+
+const {CommonListView} = goog.require('finscholar.commonlistview');
+const {ScholarshipListDataHandler} = goog.require('datahandlers.scholarshiplistdatahandler');
+
+const SCHOLARSHIP_LIST_INDEX = '1';
 
 /** The mini controller for scholarship list view. */
-class ScholarshipListView {
+class ScholarshipListView extends CommonListView {
   
   constructor() {
-    // TODO: In MVP, we'll add member variables such as pages, the array of scholarship etc.
+    super(new ScholarshipListDataHandler(), SCHOLARSHIP_LIST_INDEX);
   }
 
   /**
    * Renders a scholarship list view to the container.
    * @param {!Element} container The HTML container to load the view.
    */
-  renderView(container) {
-    container.innerHTML = scholarshiplist();
+  async renderView(container) {
+    try {
+      await super.init(container);
```

Comment:
I'm a little confused by this. every time you call the render function, you initialize everything? An init method should only be called once. It also doesn't seem like you need this method at all. Can't clients just call init()?

Reply:
Oh I didn't think about the semantic of my function calls. Yeah I deleted the method.